temporal-sdk/src/test/java/io/temporal/workflow/updateTest/UpdateWithStartTest.java

```diff
@@ -288,6 +294,41 @@ public void startVariousProcs() throws ExecutionException, InterruptedException
     assertEquals("6", handle6.getResultAsync().get());
   }
 
+  @Test
+  public void timeoutError() {
+    testWorkflowRule.getTestEnvironment().shutdownNow();
+    testWorkflowRule.getTestEnvironment().awaitTermination(5, TimeUnit.SECONDS);
+
+    WorkflowClient workflowClient = testWorkflowRule.getWorkflowClient();
+    String workflowId = UUID.randomUUID().toString();
+    WorkflowOptions options =
+        WorkflowOptions.newBuilder()
+            .setTaskQueue(testWorkflowRule.getTaskQueue())
+            .setWorkflowId(workflowId)
+            .build();
+    TestWorkflows.WorkflowWithUpdate workflow =
+        workflowClient.newWorkflowStub(TestWorkflows.WorkflowWithUpdate.class, options);
+    UpdateWithStartWorkflowOperation<String> updateOp =
+        UpdateWithStartWorkflowOperation.newBuilder(workflow::update, 0, "Hello Update")
+            .setWaitForStage(WorkflowUpdateStage.COMPLETED)
+            .build();
+
+    final AtomicReference<WorkflowServiceException> exception = new AtomicReference<>();
+    ScheduledExecutorService scheduledExecutor = Executors.newScheduledThreadPool(1);
+    Context.current()
+        .withDeadlineAfter(500, TimeUnit.MILLISECONDS, scheduledExecutor)
+        .run(
+            () ->
+                exception.set(
+                    assertThrows(
+                        WorkflowServiceException.class,
```

Comment:
The equivalent Update test receives the `WorkflowUpdateTimeoutOrCancelledException` directly here. I'm not sure why here it's wrapped inside a `WorkflowServiceException`.